### PR TITLE
Fix LDAP and Python 3.7 issues

### PR DESCRIPTION
**Pull request by dvvanessastoiber**
Fixes https://github.com/Caleydo/tdp_bi_bioinfodb/issues/1051

### Summary

* adapt _config.json_ for local testing
* fix remaining issues with Python 3.7, remove type `bytes` with `connection.response_to_json`
* change parameter in `_get_user_groups()` in _ldap.py_ to retrieve correct values for login
* change `bind_user_nd` to correct `bind_user_dn`

### Result
* Login works again

### CODE REVIEW

**Comment by thinkh** [comment]
While testing and reverting the config.json to it's original values from the develop branch, I noticed that a server with an empty hostname (`"hostname": "",`) is added to the server pool. When trying to establish the connection (in `_make_connection()`) the server cannot be found. Furthermore, the connection is stuck in an infinite loop, which results in the pending login API request.

Digging in the ldap3 documentation I found the [following passage](https://ldap3.readthedocs.io/server.html#server-pool):

> The active and exhaust parameter accept either a boolean or a number: if you set `active=True` while defining the ServerPool the strategy will check for server availability, you can also set this attribute to the maximum number of cycles to try before giving up with an _LDAPServerPoolExhaustedError_ exception. 

So I changed our server pool as follows:

```py
self._server_pool = ldap3.ServerPool([], ldap3.FIRST, active=1, exhaust=True)
```

Now the ldap connection returns the expected _LDAPServerPoolExhaustedError_ exception after a few seconds and our login API request fails with a _401 UNAUTHORIZED_ error.

In conclusion a LDAP server must be available to avoid an infinite loop!

File: phovea_security_store_ldap/main.py [hunk 1]
```diff
@@ -6,7 +6,7 @@ def _loop():
 
     log.setLevel(logging.DEBUG)
     store = LDAPStore()
-    username = input('username: ')
+    username = eval(input('username: '))
```
Comment (thinkh) [inline phovea_security_store_ldap/main.py]: Why do we need the `eval()` here? In conjunction with `input()` this is a [security hole](https://stackoverflow.com/a/9384005)!

File: phovea_security_store_ldap/main.py [hunk 2]
```diff
@@ -15,6 +15,6 @@ def _loop():
         print(('OK {u.id} name:{u.name} roles:{u.roles} dn:{u.dn} {u.info} {u.groups}'.format(u=user)))
       else:
         print('ERROR')
-      username = input('another username: ')
+      username = eval(input('another username: '))
```
Comment (thinkh) [inline phovea_security_store_ldap/main.py]: See previous comment.

**Comment by thinkh** [comment]
Other than the `eval(input())` the changes look fine.

File: phovea_security_store_ldap/main.py (hunk 1 above)
Comment (dvvanessastoiber) [reply phovea_security_store_ldap/main.py]: Thanks for making me aware of that. It has been added by the `2to3` library when converting to the new syntax and I have removed it again.

Comment (dvvanessastoiber) [reply phovea_security_store_ldap/main.py]: The former`raw_input()` (Python 2.x) has been renamed to  `input()` in Python 3.x. `eval()` has been added to provide the functionality that `input()` provided in 2.x version of python but it is not necessary in this case since it would only be necessary if the code in Python 2.x would have been `eval(raw_input())`
